Write docs on AuthSigs, SessionSigs, PKPs (#66)
LIT-Protocol/docs: docs/pkp/minting.md

# Minting A PKP

## Mint via Contracts

> You can mint a PKP NFT from our PKP contract on Polygon Mumbai [here](https://explorer.litprotocol.com/mint-pkp). This NFT represents the root ownership of the PKP. The NFT owner can grant other users (via a wallet address) or grant Lit Actions the ability to use the PKP to sign and decrypt data. They also have the ability to assign additional authentication methods, described at the bottom of the page.

Reviewer: Updating this to chronicle:

```suggestion
You can mint an NFT from our PKP contract on Chronicle, Lit's custom EVM rollup testnet, [here](https://explorer.litprotocol.com/mint-pkp). This NFT represents the root ownership of the PKP. The NFT owner can grant other users (via a wallet address) or grant Lit Actions the ability to use the PKP to sign and decrypt data. They also have the ability to assign additional authentication methods, described at the bottom of the page.

You can also use our handy auth helper contract on Chronicle [here](https://github.com/LIT-Protocol/LitNodeContracts/blob/main/contracts/PKPHelper.sol) and you can find the contract addresses [here](https://explorer.litprotocol.com/contracts).
```

You can also use our handy auth helper contract on Polygon Mumbai [here](https://github.com/LIT-Protocol/LitNodeContracts/blob/main/contracts/PKPHelper.sol) and you can find the contract addresses [here](https://explorer.litprotocol.com/contracts)

## Mint via Social

You can mint a PKP by presenting a valid OAuth token as an authentication method to the Lit Relay server. Currently, only Google OAuth tokens are supported, but we plan to support Discord in the near term. 

Read more about this process [here](/pkp/wallets/examples).

## Mint via WebAuthn

You can mint a PKP by presenting a valid WebAuthn credential generated by your browser to the Lit Relay server. 

We have a frontend that helps with this process at https://pkp-walletconnect.vercel.app/.

We currently support both username-based and username-less WebAuthn registration, and usernames are purely used for your convenience / reference on the client-side.

## Technical Details

### Contract Specifics

- The `authMethodId` is derived from the credential's [rawId](https://www.w3.org/TR/webauthn-2/#dom-publickeycredential-rawid) parameter.

> - The `authMethodPubkey` is the COSE credential public key.

Reviewer 2: should there be something about how our nodes can only support validating with ECDSA keys?

> - The `authMethodPubkey` is the COSE credential public key. **We currently only support Elliptic Curve COSE Key Type IDs**.

Reviewer: ```suggestion
- The `authMethodPubkey` is the [COSE credential public key](https://datatracker.ietf.org/doc/html/rfc8812). **We currently only support Elliptic Curve COSE Key Type IDs**.
```

link out to COSE and additional reading about it

### Relying Parties and Supported Origins

> In order to allow for various frontends to integrate with our platform, we plan to support any domain to act as a [Relying Party](https://www.w3.org/TR/webauthn-2/#webauthn-relying-party) in the long run. However, we are in the process of slowly rolling out this authentication method currently maintain an allowlist of origins / domains that can integrate with the Lit network.

Reviewer 2: Should we mention that nodes will require `localhost:3000` to be the `rpId`

Author: think we removed the origin allowlist criteria, and RP can be anything now.

### Challenge-Free Registration

> 

Reviewer: What does a challenge mean? Can you add a definition.

> We do not currently use challenges as part of our PKP minting / WebAuthn registration process and only use it for the PKP / WebAuthn authentication step.

Reviewer: let's also link to the glossary for challenges, since this is the first time mentioned if someone comes to this page 
```suggestion
We do not currently use [challenges](/resources/glossary#challenge) as part of our PKP minting / WebAuthn registration process and only use it for the PKP / WebAuthn authentication step.
```
